packages/x-data-grid/src/hooks/features/listView/useGridListView.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import type { GridStateColDef } from '../../../models/colDef/gridColDef';
+import { GridStateInitializer } from '../../utils/useGridInitializeState';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridPrivateApiCommunity } from '../../../models/api/gridApiCommunity';
+import { GridListViewApi } from '../../../models/api/gridListViewApi';
+import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { GridEventListener } from '../../../models/events';
+import {
+  gridListColumnSelector,
+  gridVisibleListColumnDefinitionsSelector,
+} from './gridListViewSelectors';
+import { gridColumnDefinitionsSelector, gridVisibleColumnDefinitionsSelector } from '../columns';
+import { gridDimensionsSelector } from '../dimensions';
+import { useGridApiEventHandler } from '../../utils/useGridApiEventHandler';
+
+export type GridListViewState = { listColumn: GridStateColDef | undefined };
+
+const getListColumnWidth = (apiRef: React.MutableRefObject<GridPrivateApiCommunity>) => {
```

Comment:
Lint: if you use `function` instead of `const`, this function can be defined at the end of the file. This style makes all our hooks/features start with initializer, then hook, then helpers. Makes it easier to read, imho.